opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -1965,43 +1977,76 @@ const completeSpecialFilterKeys = async (context, user, inputFilters) => {
+        }
+        if (newFilterGroup) {
+          finalFilterGroups.push(newFilterGroup);
+        }
+      }
+      if (arrayKeys.includes(WORKFLOW_FILTER) || arrayKeys.includes(X_OPENCTI_WORKFLOW_ID)) {
+        // in case we want to filter by status template (template of a workflow status) or status
+        // we need to find all statuses filtered by status template and filter on these statuses
+        const newFilter = await adaptFilterToWorkflowFilterKey(context, user, filter);
         finalFilters.push(newFilter);
       }
-      if (newFilterGroup) {
-        finalFilterGroups.push(newFilterGroup);
+      if (arrayKeys.includes(SOURCE_RELIABILITY_FILTER)) {
+        // in case we want to filter by source reliability (reliability of author)
+        // we need to find all authors filtered by reliability and filter on these authors
+        const { newFilter, newFilterGroup } = await adaptFilterToSourceReliabilityFilterKey(context, user, filter);
+        if (newFilter) {
+          finalFilters.push(newFilter);
+        }
+        if (newFilterGroup) {
+          finalFilterGroups.push(newFilterGroup);
+        }
       }
-    } else if (arrayKeys.includes(SOURCE_RELIABILITY_FILTER)) {
-      // in case we want to filter by source reliability (reliability of author)
-      // we need to find all authors filtered by reliability and filter on these authors
-      const { newFilter, newFilterGroup } = await adaptFilterToSourceReliabilityFilterKey(context, user, filter);
-      if (newFilter) {
-        finalFilters.push(newFilter);
+      if (arrayKeys.includes(INSTANCE_FILTER)) {
+        const nested = [{ key: 'internal_id', operator: filter.operator, values: filter.values }];
+        finalFilters.push({ key: 'connections', nested });
+      }
+      if (arrayKeys.includes(RELATION_FROM_FILTER) || arrayKeys.includes(RELATION_TO_FILTER)
+          || arrayKeys.includes(RELATION_TO_SIGHTING_FILTER)) {
+        const side = arrayKeys.includes(RELATION_FROM_FILTER) ? 'from' : 'to';
+        const nested = [
+          { key: 'internal_id', operator: filter.operator, values: filter.values },
+          { key: 'role', operator: 'wildcard', values: [`*_${side}`] }
+        ];
+        finalFilters.push({ key: 'connections', nested });
+      }
+      if (arrayKeys.includes(RELATION_FROM_TYPES_FILTER) || arrayKeys.includes(RELATION_TO_TYPES_FILTER)) {
+        const side = arrayKeys.includes(RELATION_FROM_TYPES_FILTER) ? 'from' : 'to';
+        const nested = [
+          { key: 'types', operator: filter.operator, values: filter.values },
+          { key: 'role', operator: 'wildcard', values: [`*_${side}`] }
+        ];
+        finalFilters.push({ key: 'connections', nested });
       }
-      if (newFilterGroup) {
-        finalFilterGroups.push(newFilterGroup);
+      if (arrayKeys.includes(INSTANCE_FILTER_TARGET_TYPES)) {
+        const nested = [{ key: 'types', operator: filter.operator, values: filter.values }];
```

Comment:
in buildRelationsFilter, if we also have a filter on elementId, we also add in the nested : { key: 'internal_id', values: optsElementIds, operator: FilterOperator.NotEq }